rgo/git.py

```diff
@@ -167,6 +167,54 @@ def describe(self, pkg=None):
         version = version.replace("-", "_")
         return version, release
 
+    def _check_output(self, cmd):
+        env = dict(os.environ)
+        env["LC_ALL"] = "C.UTF-8"
+        return subprocess.check_output(cmd, cwd=self.cwd, encoding="utf-8", env=env).strip().splitlines()
+
+    def get_describe_long(self, ref):
+        """
+        Run git describe --tags --long <ref>
+        Return the a tuple with the following data:
+          * desc: raw `git desribe --tags --long <ref>` output
+          * tag: last available tag in commits <= ref (parsed from desc)
+          * commits: number of commits between the tag and the ref (parsed from desc)
+          * hash: hash of the ref (parsed from desc)
+        """
+        cmd = ["git", "describe", "--tags", "--long", ref]
+        desc = self._check_output(cmd)[0].strip()
```

Comment:
Come think of it, what happens if there are no tags? We should probably check the describe returned what we expected?

Reply:
(this is resolved in the other PR, might be nicer to have it handled here, but no biggie)

Reply:
There's a fallback to `git rev-list --count` now